src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -202,8 +223,6 @@ private bool IsOpenProperty(ODataProperty property)
 
             WriteInstanceAnnotation(property, isTopLevel, currentPropertyInfo.MetadataType.IsUndeclaredProperty);

```

Comment:
this logic should come before the check for null values.  Even if the property is null, it can have instance annotations (in fact, a common use case for instance annotations is annotating a null value with why it is null).  Please move this before the check for null values, and add a test to verify that instance annotations are correctly written/read for annotations applied to null values, with and without omit-nulls.

Reply:
Moved. Thanks.


---
In reply to: [186558582](https://github.com/OData/odata.net/pull/1106#discussion_r186558582) [](ancestors = 186558582)